src/extensions/lib/src/check-firmware-versions.ts

```diff
@@ -0,0 +1,69 @@
+import type { ApiExtensionContext } from '@directus/extensions';
+
+type ProbeInfo = {
+	id: string;
+	ip: string;
+	name: string | null;
+	hardwareDeviceFirmware: string | null;
+	nodeVersion: string | null;
+}
+
+export const getFirmwareSubject = (version: string) => `Outdated probe firmware, should be ${version}`;
+export const getNodeVersionSubject = (version: string) => `Outdated probe node.js version, should be ${version}`;
```

Comment:
"Your probe is running an outdated firmware" in both cases since, for the users, there's no difference between node/FW; they always just update the FW. Also no version in the title, I suppose you can use the text for "already sent" detection for now.

Reply:
Having version in the subject was a hack to filter out already sent notifications for another version. Since we will change the duplicates search approach, that may be just text.